src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,50 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        if (movie == null) {
+            throw new RuntimeException("unacceptable data");
+        }
+
+        if (movie.getId() != null && get(movie.getId()).isPresent()) {
+            throw new RuntimeException(movie + "already existed in database");
+        }
+
+        Session session = null;
+        Transaction transaction = null;
+
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingExeption("adding " + movie + " into database failed");
```

Comment:
The exception thrown here uses `DataProcessingExeption`, which contains a typo. It should be `DataProcessingException` to ensure the correct exception class is used.